src/core/libmaven/mzSample.cpp

```diff
@@ -963,10 +968,24 @@ void mzSample::parseMzXMLScan(const xml_node &scan, const int& scannum)
 		scanpolarity = getPolarityFromfilterLine(filterLine);
 	}
 
-	//TODO: What is this for this is zero
-	precursorMz = string2float(string(scan.child_value("precursorMz")));
-
-	// string b64String; naman Unused variable: b64String
+    xml_node precursor = mzxml_scan_node.child("precursorMz");
+    if (precursor) {
+        _scan->precursorMz = string2float(mzxml_scan_node.child_value("precursorMz"));
```

Comment:
The variable `_scan` is also not known in this scope.

Reply:
Fixing it. Lost my changes once and reproduced it from memory. Forgot a few crucial things. My bad.